### PR TITLE
325: Split the user settings and profile in to separate pages.

### LINKED ISSUE #325: Public and private profiles do not have the same design
If you login and select your profile link (your private profile), you get your settings page.  However if you go to your public profile you get your avatar, description, recent projects and what validator rights you have.

Your private profile should have the same information as your public one.

### PR DESCRIPTION
**Pull request by toolstack**
Alternative solution to resolve #325.

### CODE REVIEW

**Comment by toolstack** [comment]
Screen shots:

Profile screen, available at /glotpress/profile or /glotpress/profile/$user:
profile screen

Settings screen, available at /glotpress/settings:
settings screen

File: gp-includes/template.php [hunk 1]
```diff
@@ -105,7 +105,10 @@ function gp_nav_menu_items( $location = 'main' ) {
 	}
 	elseif ( 'side' === $location ) {
 		if ( is_user_logged_in() ) {
-			$items[ gp_url( '/profile' ) ] = __( 'Profile', 'glotpress' );
+			$user = wp_get_current_user();
+			$login = esc_html( $user->user_login );
+			$items[ gp_url( '/profile/' . $login ) ] = __( 'Profile', 'glotpress' );
```
Comment (ocean90) [inline gp-includes/template.php]: This should use `gp_url_profile()`. Note that the slug of a user is `user_nicename`, not `user_login`.

Comment (toolstack) [inline gp-includes/template.php]: Updated.

File: gp-includes/routes/settings.php [hunk 2]
```diff
@@ -0,0 +1,43 @@
+<?php
+/**
+ * @method object|array many_no_map()
+ * @method object|array find_many_no_map()
+ */
+class GP_Route_Settings extends GP_Route_Main {
+	// For caching purposes
+	private $projects = array();
```
Comment (ocean90) [inline gp-includes/routes/settings.php]: This seems to be unused.

File: gp-includes/routes/settings.php [hunk 3]
```diff
@@ -0,0 +1,43 @@
+<?php
+/**
+ * @method object|array many_no_map()
+ * @method object|array find_many_no_map()
+ */
```
Comment (ocean90) [inline gp-includes/routes/settings.php:14]: Hmm, GP_Route_Settings, GP_Route_Main and GP_Router don't support this methods?

Comment (toolstack) [inline gp-includes/routes/settings.php:14]: Old code for scrutinizer, no longer required.  I'll remove it.

File: gp-includes/routes/settings.php (hunk 2 above)
Comment (toolstack) [inline gp-includes/routes/settings.php]: Agreed, I'll remove.

File: gp-includes/routes/profile.php [hunk 4]
```diff
@@ -16,34 +16,13 @@ public function profile_get() {
...
-			$user_sort = wp_parse_args( $_POST['default_sort'], $default_sort );
-			update_user_option( $user_id, 'gp_default_sort', $user_sort );
+	public function profile_view( $user = '' ) {
+		if ( '' == $user ) { 
+			$user = wp_get_current_user(); 
+		} else {		
+			$user = get_user_by( 'slug', $user );
 		}
 
-		$this->redirect( gp_url( '/profile' ) );
```
Comment (ocean90) [inline gp-includes/routes/profile.php:50]: IMO `/profile` should redirect to `/profile/$current_user`. In case of a non-logged in access return a 404.

**Comment by ocean90** [comment]
I'm seeing a few CS issues here.
- `WordPress.XSS.EscapeOutput.UnsafePrintingFunction` can be ignored, see #340.
- Whitespace issues should be fixed in this PR.
- Missing nonce verification can be handled in an extra PR.

**Comment by ocean90** [comment]
@toolstack For issues like `Expected next thing to be an escaping function (see Codex for 'Data Validation'), not 'gp_radio_buttons' (WordPress.XSS.EscapeOutput.OutputNotEscaped)` you can add a comment `//xss ok`, see https://github.com/WordPress-Coding-Standards/WordPress-Coding-Standards/blob/e3b8e576a09bcea0ca9eae713e2d822e438c66e8/WordPress/Sniffs/XSS/EscapeOutputSniff.php#L146

**Comment by toolstack** [comment]
Funny story, if you use `//xss ok` inline it throws multiple coding standard errors (no space after the slashes, no capitalization and no ending punctuation) :wink:

**Comment by ocean90** [comment]
@toolstack Sorry, the comment is `// WPCS: xss ok.`, see https://github.com/WordPress-Coding-Standards/WordPress-Coding-Standards/wiki/Whitelisting-code-which-flags-errors#escaping--xss.

**Pull request merged**